casbin/enforcer.cpp

```diff
@@ -200,7 +201,7 @@ Enforcer :: Enforcer(shared_ptr<Model> m, shared_ptr<Adapter> adapter) {
 
     this->Initialize();
 
-    if (this->adapter->file_path != "") {
+    if (this->adapter && this->adapter->file_path != "") {
```

Comment:
I think no need to check `file_path` here, because every `Adapter` has a `LoadPolicy` method.

Reply:
but adapter is NULLABLE. or you should remove this constructor to avoid misleading users.

```cxx
/**
 * Enforcer initializes an enforcer with a model.
 *
 * @param m the model.
 */
Enforcer ::Enforcer(shared_ptr<Model> m): Enforcer(m, NULL) {
}
```

Reply:
Adding `this->adapter &&` null check is OK.